package-res/resources/web/pentaho/type/property.js

```diff
@@ -0,0 +1,1331 @@
+
+        /**
+         * Gets or sets the name of the _property type_.
+         *
+         * The name of a _property type_ identifies it within
+         * its [declaring type]{@link pentaho.type.Property.Type#declaringType}.
+         *
+         * ### Set
+         *
+         * This attribute must be set when defining a new _property type_,
+         * and cannot be changed afterwards.
+         *
+         * When set to a non-{@link Nully} and non-{@link String} value,
+         * the value is first replaced by the result of calling its `toString` method.
+         *
+         * @type {!nonEmptyString}
+         *
+         * @throws {pentaho.lang.ArgumentRequiredError} When set to an empty string or a _nully_ value.
+         * @throws {TypeError} When set to a value different from the current one.
+         *
+         * @see pentaho.type.spec.IPropertyTypeProto#name
+         */
+        get name() {
+          return this._name;
+        },
+
+        set name(value) {
+          value = nonEmptyString(value);
+
+          if(!value) throw error.argRequired("name");
+
+          // Only stored at the root property type.
+          if(this.isRoot) {
+            // Cannot change, or throws.
+            O.setConst(this, "_name", value);
+          } else {
+            // Hierarchy consistency
+            if(value && value !== this._name)
+              throw new TypeError("Sub-properties cannot change the 'name' attribute.");
+          }
+        }, //endregion
+
+        //region list attribute
+        /**
+         * Gets a value that indicates if the property is a _list_.
+         *
+         * A property is a _list_ property if
+         * its [value type]{@link pentaho.type.Property.Type#type} is a list type,
+         * i.e., if it is or extends [List]{@link pentaho.type.List}.
+         *
+         * @type boolean
+         * @readonly
+         */
+        get isList() {
+          return this._type.isList;
+        }, //endregion
+
+        //region (value) element type attribute
+        /**
+         * The base element type of the _singular_ values that the property can hold.
+         *
+         * When the property is a list property,
+         * that list type's element type,
+         * {@link pentaho.type.List.Type#of},
+         * is returned.
+         *
+         * Otherwise,
+         * {@link pentaho.type.Property.type} is returned.
+         *
+         * @type !pentaho.type.Element.Type
+         * @readonly
+         */
+        get elemType() {
+          var type = this._type;
+          return type.isList ? type.of : type;
+        }, //endregion
+
+        //region (value) type attribute
+        _type: undefined,
+
+        /**
+         * Gets or sets the type of value that properties of this type can hold.
+         *
+         * If the _value type_ is a [list]{@link pentaho.type.Value.Type#isList} type,
+         * then this property will be a _list_ (or multiple-elements) property,
```

Comment:
Suggest:

then this property will be a _list_ (or multiple-elements) property;